akka-docs/src/main/paradox/typed/cluster-membership.md

```diff
@@ -115,35 +115,30 @@ members in quorum decisions.
 
 ## State Diagrams
 
-### State Diagram for the Member States (`akka.cluster.allow-weakly-up-members=off`)
+### State Diagram for the Member States 
 
-![member-states.png](../images/member-states.png)
-
-### State Diagram for the Member States (`akka.cluster.allow-weakly-up-members=on`)
-
-![member-states-weakly-up.png](../images/member-states-weakly-up.png)
+![member-states.png](../images/member-state-diagram.png)
   
 #### User Actions
 
  * **join** - join a single node to a cluster - can be explicit or automatic on
 startup if a node to join have been specified in the configuration
    
- * **leave** - tell a node to leave the cluster gracefully
+ * **leave** - tell a node to leave the cluster gracefully, normally triggered by ActorSystem or JVM shutdown through @ref[coordinated shutdown](../coordinated-shutdown.md)
    
- * **down** - mark a node as down
+ * **down** - mark a node as down - normally only used to remove crashed nodes from the cluster
    
 #### Leader Actions
 
-The `leader` has the following duties:
+The `leader` has the following duties of shifting members in and out of the cluster:
 
- * shifting members in and out of the cluster
-    * joining -> up
-    * weakly up -> up *(no convergence is required for this leader action to be performed)*
-    * exiting -> removed
+ * joining ⭢ up
+ * weakly up ⭢ up *(no convergence is required for this leader action to be performed)*
```

Comment:
Is that even correct? Shouldn't there be another edge "joining -> weakly up *(no convergence is required for this leader action to be performed)*". The edge "weakly up -> up" needs full convergence AFAIK.

```suggestion
 * joining ⭢ weakly up *(only gossip convergence is required for this leader action to be performed which works even if there are unreachable nodes)*
 * weakly up ⭢ up *(will be performed after full convergence is reached again)*
```

Reply:
No, that's right and is the whole point of weakly up, to allow nodes to work for cluster in the face of unreachability.

Reply:
Exactly - so you can become 'weakly up' and start doing some work without convergence, but not fully 'up', right? https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala#L1155

Reply:
Sorry, yes my suggestion was a weird mix of things. I think the current text is still wrong as well :) Here's a new suggestion after actually reading the code.

```suggestion
 * joining ⭢ weakly up *(no convergence is needed for this leader action to be performed which works even if there are unreachable nodes)*
 * weakly up ⭢ up *(will be performed after full convergence is reached again)*
```